warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -729,6 +730,18 @@ protected ASTJexlScript updateQueryTree(ScannerFactory scannerFactory, MetadataH
         
         stopwatch.stop();
         
+        // Fix query property markers with multiple sources.
+        stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - fix query property markers with multiple sources");
+        try {
+            queryTree = QueryPropertyMarkerSourceConsolidator.consolidate(queryTree);
+        } catch (Exception e) {
+            throw new DatawaveQueryException("Failed to fix query property markers with multiple sources", e);
+        }
+        if (log.isDebugEnabled()) {
+            logQuery(queryTree, "Query after fixing query property markers with multiple sources");
+        }
+        stopwatch.stop();
+        
```

Comment:
If #1274 is merged before this one, extract this to a `timedFixQueryPropertyMarkers()` method call

Reply:
Okay, no problem.